src/Grids/latitude_longitude_grid.jl

```diff
@@ -549,11 +539,14 @@ function allocate_metrics(grid::LatitudeLongitudeGrid)
         metric_size = (length(grid.Δλᶜᵃᵃ)   , length(grid.φᵃᶜᵃ))
     end
 
-    for metric in grid_metrics
-        parentM        = Symbol(metric, :_parent)
-        @eval $parentM = zeros($FT, $metric_size...)
-        @eval $metric  = OffsetArray(on_architecture($arch, $parentM), $offsets...)
-    end
+    Δxᶠᶜ = OffsetArray(zeros(FT, arch, metric_size...), offsets...)
```

Comment:
Now I am questioning the use of `eval` from the beginning :)

You were defining new global variables.

Reply:
what was the purpose of the `@eval`?